purchase_order_analytic_search/models/purchase_order.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Eficent Business and IT Consulting Services S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+
+    _inherit = "purchase.order"
+
+    @api.multi
+    @api.depends('order_line.account_analytic_id')
+    def _compute_analytic_accounts(self):
+        res = {}
+        for purchase in self:
+            res[purchase.id] = []
+            for po_line in purchase.order_line:
+                if po_line.account_analytic_id:
+                    res[purchase.id].append(po_line.account_analytic_id.id)
+        return res
```

Comment:
this is not the way to define a computed field in the new api. You need to assign a recordset to purchase.analytic_account_ids.